libdebug/ptrace/ptrace_interface.py

```diff
@@ -350,6 +350,55 @@ def finish(self: PtraceInterface, thread: ThreadContext, heuristic: str) -> None
         else:
             raise ValueError(f"Unimplemented heuristic {heuristic}")
 
+    def next(self: PtraceInterface, thread: ThreadContext) -> None:
+        """Executes the next instruction of the process. If the instruction is a call, the debugger will continue until the function returns.
+        """
+
+        # Check if the current instruction is a call and its skip
+        is_call = self.lib_trace.is_call_instruction(
```

Comment:
Do these functions need to be in the CFFI implementation? We could simply have them here in the Python code, calling the native code might be slower than performing a few bitwise checks from Python

Reply:
CFFI:

Ran 30000 tests in 1091.935s

OK
All tests passed
Total time: 1094.32s

____________________________________

Ran 30000 tests in 1376.933s

OK
All tests passed
Total time: 1379.57s

____________________________________

Python is 26.1% slower

Reply:
Still, this would only matter in extensive bruteforces.

Reply:
Still, given discussion, I'm going to keep the Python version.

Reply:
Resolved in commit [970765d](https://github.com/libdebug/libdebug/commit/970765d9fb761638831e422cc9cb8a850ce8d525)